lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -71,8 +72,10 @@ def date_range(self, start: AnyStr, end: AnyStr) -> list:
         date_range = []
         start_date = date_parser.parse(start)
         end_date = date_parser.parse(end)
+        first_month = start_date.replace(day=1)
+        last_month = end_date + relativedelta(day=31)
         assert start_date <= end_date
-        for m in rrule.rrule(rrule.MONTHLY, dtstart=start_date, until=end_date):
+        for m in rrule.rrule(rrule.MONTHLY, dtstart=first_month, until=last_month):
```

Comment:
The assertion is either in the wrong place (e.g., it should be positioned before the values of `start_date` and `end_date` are used), or it's testing the wrong stuff (i.e., it should be testing `first_month` and `last_month`, instead).  (I vote for the latter....)

Reply:
I patched this independently of the fix that's now on main, just to get it running. This'll be made consistent (with the assert first) once I rebase.